gcs/src/main/scala/io/monix/connect/gcs/configuration/BlobInfo.scala

```diff
@@ -0,0 +1,148 @@
+package io.monix.connect.gcs.configuration
+
+import java.time.Instant
+
+import com.google.cloud.storage.BlobInfo.CustomerEncryption
+import com.google.cloud.storage.{Acl, BlobId, BlobInfo => GoogleBlobInfo, StorageClass}
+
+import scala.jdk.CollectionConverters._
+
+final case class BlobInfo(name: String,
+                          bucket: String,
+                          generatedId: String,
+                          cacheControl: Option[String],
+                          size: Long,
+                          contentType: Option[String],
+                          contentEncoding: Option[String],
+                          contentDisposition: Option[String],
+                          contentLanguage: Option[String],
+                          componentCount: Int,
+                          etag: Option[String],
+                          md5: Option[String],
+                          md5ToHexString: Option[String],
+                          crc32c: Option[String],
+                          crc32cToHexString: Option[String],
+                          mediaLink: Option[String],
+                          metadata: Map[String, String],
+                          generation: Long,
+                          metageneration: Long,
+                          deleteTime: Instant,
+                          updateTime: Instant,
+                          createTime: Instant,
+                          isDirectory: Boolean,
+                          customerEncryption: Option[CustomerEncryption],
+                          storageClass: StorageClass,
+                          kmsKeyName: String,
+                          eventBasedHold: Option[Boolean],
+                          temporaryHold: Option[Boolean],
+                          retentionExpirationTime: Option[Instant])
+
+
+object BlobInfo {
+
+
+  final case class Create(name: String,
+                          contentType: Option[String] = None,
+                          contentDisposition: Option[String] = None,
+                          contentLanguage: Option[String] = None,
+                          contentEncoding: Option[String] = None,
+                          cacheControl: Option[String] = None,
+                          crc32c: Option[String] = None,
+                          crc32cFromHexString: Option[String] = None,
+                          md5: Option[String] = None,
+                          md5FromHexString: Option[String] = None,
+                          metadata: Map[String, String] = Map.empty[String, String],
+                          storageClass: Option[StorageClass] = None,
+                          acl: List[Acl] = List.empty[Acl],
+                          eventBasedHold: Option[Boolean] = None,
+                          temporaryHold: Option[Boolean] = None)
+
+  final case class Update(contentType: Option[String] = None,
```

Comment:
`Create` and `Update` look the same but for the first field, maybe we can use a generic abstraction to join them?

Reply:
I wanted to abstract over the ACL methods inside of Bucket and Blob as well, as they're the same, but couldn't figure out how to, as they don't seem to use a common interface. I need to double check this.

It would be nice to be able to abstract over the CRUD methods on Bucket and Blob similarly.